tools/transform.cpp

```diff
@@ -94,7 +94,7 @@ static void print_varval(ostream &os, State &st, const Model &m,
   for (unsigned i = 0, e = agg->numElementsConst(); i < e; ++i) {
     if (i != 0)
       os << ", ";
-    print_varval(os, st, m, var, agg->getChild(i), agg->extract(val, i));
+    print_varval(os, st, m, nullptr, agg->getChild(i), agg->extract(val, i));
```

Comment:
Fixed here because `var` cannot be used when printing agg's elements.
It makes print_single_varval's `assert(n == 0)` fail.

Reply:
I think the right fix is to skip padding elements

Reply:
I tried this, and it still happened when -disable-undef-input was added.
I updated print_single_varval to deal with the case when n was 2:
```
    if (n == 1) {
      os << "undef";
      return;
    } else if (n == 2) { // this branch is added
      os << "poison";
      return;
    }
```